File: lua/examples/guards/scripts/Beacon.lua
---@uuid 7c259fd2-fdb9-453b-a7ef-19cdd85428cc
---@class Beacon : Script
Beacon = script_class()

function Beacon:on_update()
    local pos = self.node.global_position
    table.insert(Plugin:get("Game").beacons, pos)
    print("beacon registered: " .. tostring(self.node));
    self.node:destroy()
end

File: cs/examples/Guards/Beacon.cs
using FyroxLite;

[Uuid("7c259fd2-fdb9-453b-a7ef-19cdd85428cc")]
public class Beacon : NodeScript
{
    protected override void OnUpdate(float dt)
    {
        Plugin.Get<Game>().beacons.Add(Node.GlobalPosition);

        Log.Info($"Beacon registered: {Node}");

        Node.Destroy();
    }
}